src/Framework/ParticleData/BaryonResUtils.cxx

```diff
@@ -20,557 +52,817 @@
-                         assert(false);
-    }
-    return false;
+  switch(res) {
+
+  case kP33_1232:  return true;  break;
+  case kS11_1535:  return false; break;
+  case kD13_1520:  return false; break;
+  case kS11_1650:  return false; break;
+  case kD13_1700:  return false; break;
+  case kD15_1675:  return false; break;
+  case kS31_1620:  return true;  break;
+  case kD33_1700:  return true;  break;
+  case kP11_1440:  return false; break;
+  case kP33_1600:  return true;  break;
+  case kP13_1720:  return false; break;
+  case kF15_1680:  return false; break;
+  case kP31_1910:  return true;  break;
+  case kP33_1920:  return true;  break;
+  case kF35_1905:  return true;  break;
+  case kF37_1950:  return true;  break;
+  case kP11_1710:  return false; break;
+  case kF17_1970:  return false; break;
+  default:
+    // should not be here - meaningless to return anything
+    gAbortingInErr = true;
+    LOG("BaryonResUtils", pFATAL) << "Unknown resonance " << res;
+    exit(1);
+  }
+  return false;
 }
 //____________________________________________________________________________
-// The values of resonance mass and width is taken from
+// The values of resonance mass and width is taken from 
 // M. Tanabashi et al. (Particle Data Group) Phys. Rev. D 98, 030001
 bool genie::utils::res::IsN(Resonance_t res)
 {
-    return (! utils::res::IsDelta(res) );
+  return (! utils::res::IsDelta(res) );
 }
 //____________________________________________________________________________
 double genie::utils::res::Mass(Resonance_t res)
 {
-    switch(res) {
-        case kP33_1232  : return 1.232 * units::GeV ; break;
-        case kS11_1535  : return 1.535 * units::GeV ; break;
-        case kD13_1520  : return 1.515 * units::GeV ; break;
-        case kS11_1650  : return 1.655 * units::GeV ; break;
-        case kD13_1700  : return 1.700 * units::GeV ; break;
-        case kD15_1675  : return 1.675 * units::GeV ; break;
-        case kS31_1620  : return 1.630 * units::GeV ; break;
-        case kD33_1700  : return 1.700 * units::GeV ; break;
-        case kP11_1440  : return 1.430 * units::GeV ; break;
-        case kP33_1600  : return 1.570 * units::GeV ; break;
-        case kP13_1720  : return 1.720 * units::GeV ; break;
-        case kF15_1680  : return 1.685 * units::GeV ; break;
-        case kP31_1910  : return 1.890 * units::GeV ; break;
-        case kP33_1920  : return 1.920 * units::GeV ; break;
-        case kF35_1905  : return 1.880 * units::GeV ; break;
-        case kF37_1950  : return 1.930 * units::GeV ; break;
-        case kP11_1710  : return 1.710 * units::GeV ; break;
-        case kF17_1970  : return 2.190 * units::GeV ; break;
-        default: break;
-    }
+  if (res == kNoResonance)      
     return -1;
+
+  // Hardcoded data are removed, now they are taken from PDG table via TDatabasePDG and cached
+  static std::map<Resonance_t, double> cache ;
```

Comment:
Is this a stopgap?  Relies on a static cache and references the "Instance" of the PDG table; in the long term of reweighting where two models use different PDG table we must think about how this will be handled.

Reply:
Good question. I had a bit of time to discuss it with Costas. We realised that trying to reweight everything is borderline with silly. Especially because the framework itself needs masses and they are coming from the pDG table. In that case which mass (PDG table) do we use? There is not smart answer especially because we know the masses are not easy to reweight. 
So at the moment we think this is a no harm solution. In the future, if we deem this not enough, we can remove easily the caching. But at this stage of my understanding of the code evolution, I think it's not going to happen.